modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -130,6 +130,166 @@ Utilities::getProjectionModel(const ComplexData* data,
     }
 }
 
+six::NITFReadControl six::sicd::Utilities::getLoadedNITFReadControl(
+        const std::string& sicdPathname,
+        const std::vector<std::string>& schemaPaths)
+{
+    six::XMLControlRegistry xmlRegistry;
+    xmlRegistry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+            six::sicd::ComplexXMLControl>());
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&xmlRegistry);
```

Comment:
Eh, is this safe?  The read control is holding onto a pointer to the registry, and is obviously not taking ownership since the registry is on the stack.  Is the registry only used during the load() when it parses the XML?  If someone tried to do another load() with the returned reader, they'd get a seg fault because they'd be holding onto a pointer to an invalid registry right?  Assuming this is all true, after the load(), add a setXMLControlRegistry(NULL) w/ a comment to make this safer.

Longer term I think we want to drop the registry altogether as it's not used like it was originally intended and just makes stuff a headache.